### ISSUE TITLE
[Issue] User Management: Deactivate (soft-delete) users via the API

### ISSUE THREAD

**Issue by TylerHendrickson** [usdigitalresponse/cpf-reporter#104]
**DRAFT: THIS TICKET IS NOT YET READY FOR WORK**

**Subtask of user story:** User Management https://github.com/usdigitalresponse/cpf-reporter/issues/81

**DOD:** Users can be made inactive via the API. 

**Open questions:**
1. Is this necessary, or should `updateUser` include deactivation instead?

**Blocked by issues:** (None)

**Blocks these issues:**
- https://github.com/usdigitalresponse/cpf-reporter/issues/101
- https://github.com/usdigitalresponse/cpf-reporter/issues/102

**Implementation details:**
- Per **Open question 1**, if this mutation is not necessary, it should be removed from the `users` service and the gql schema in `users.sdl.ts`.
- Create a `deleteUser` mutation resolver which deactivates a user identified by the request when the following scenarios are satisfied:
  - The authenticated user has role `USDR_ADMIN`
  - The authenticated user has role `ORGANIZATION_ADMIN`, and the organization associated with the authenticated user matches that of the user being deleted.
  - The authenticated user is not the same user that is being deactivated (i.e. users cannot self-deactivate).
- User deactivation is actually an **update, _not_ a delete operation**, which sets the user's `isActive` column to `false`.

**Comment by TylerHendrickson** [comment]
Closing in favor of handling this in https://github.com/usdigitalresponse/cpf-reporter/issues/102
